01/assets/js/app.js

```diff
@@ -1,4 +1,49 @@
-const a = '4.2';
-const b = 9;
+let a = parseInt('4.2');
+let b = 9;
 
-console.log(a, b);
\ No newline at end of file
+// console.log(a, b);
+// console.log(typeof a, typeof b);
+
+const addition = (a, b) => {
+	return a + b;
+};
+const subtraction = (a, b) => {
+	return a - b;
+};
+const multiplication = (a, b) => {
+	return a * b;
+};
+const exponentiation = (a, b) => {
+	return a ** b;
+};
+const division = (a, b) => {
+	return a / b;
+};
+const modulus = (a, b) => {
+	return a % b;
+};
+
+const operations = {
+	'+': addition,
+	'-': subtraction,
+	'*': multiplication,
+	'**': exponentiation,
+	'/': division,
+	'%': modulus,
+};
+const checkNumber = (operationValue, operation, index) => {
+	const operationType = operation.toString();
+	const operationOperand = Object.keys(index);
+
+	operationOperand.forEach(operand => {
+		if (operationType.includes(operand)) {
+			console.log(`Result: ${operationValue}`);
+			if (operationValue >= 20) {
+				console.log(`Result of ${operand} is greater or equal 20`);
+			} else {
+				console.log(`Result of ${operand} is less than 20`);
+			}
+		}
+	});
+};
+checkNumber(operations['+'](a, b), operations['+'], operations);
```

Comment:
Teraz wygląda to zdecydowanie lepiej, ale... zauważyłem, że lubisz sobie komplikować życie/rozwiązanie.

Jeśli jest coś prostego to nie mówię, aby nie pisać funkcji, ale nie ma potrzeby bardzo komplikować rozwiązania. Ma to być elastyczne, ale staraj się nie przesadzać (wiem, że trudno określić granice). Czasem nie warto rozbudowywać funkcji bo okaże się, że nigdy już do niej nie wrócisz. Po prostu zwracam Ci uwagę na temat over-engineering:  https://geek.justjoin.it/poznaj-dobrze-domene-a-unikniesz-over-engineeringu/